src/Symfony/Component/Validator/Constraints/Image.php

```diff
@@ -192,6 +196,10 @@ public function __construct(
         $this->allowPortraitMessage = $allowPortraitMessage ?? $this->allowPortraitMessage;
         $this->corruptedMessage = $corruptedMessage ?? $this->corruptedMessage;
 
+        if (null === $this->mimeTypes && [] === $this->extensions) {
```

Comment:
doesn't the second check change the behavior? extensions is null by default, so we won't set mimeTypes to its default anymore, right?

Reply:
The property is initialised with an empty array in the parent class which then also has this line in its constructor (which is called before the code here is run):

```php
$this->extensions = $extensions ?? $this->extensions;
```